numpy/include/numpy.hpp

```diff
@@ -102,15 +102,14 @@ class ndarray {
     }
     template <typename U>
     auto binary_operator_add_impl(const U& other) const {
+        if constexpr (std::is_same_v<U, float_>) {
+            xt::xarray<float_> result = xt::cast<float_>(_array) + other;
+            return ndarray<float_>(result);
+        }
```

Comment:
you'd better use an else branch to include the rest three lines. This can prevent some unnecessary template instantiations error.

Reply:
I thought not encountering a return would lead to warnings on compilation, but you're correct that won't be the case here.